From 54ef2e904cd94d0a746bdcf9d6136a8905df2300 Mon Sep 17 00:00:00 2001
From: Ralph Sennhauser <ralph.sennhauser@gmail.com>
Date: Sun, 27 Apr 2025 17:21:54 +0200
Subject: [PATCH 01/10] Add missing header in KeyName

---
 source/ps/KeyName.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/source/ps/KeyName.cpp b/source/ps/KeyName.cpp
index 7abd91841c..1cfa06c15d 100644
--- a/source/ps/KeyName.cpp
+++ b/source/ps/KeyName.cpp
@@ -1,4 +1,4 @@
-/* Copyright (C) 2021 Wildfire Games.
+/* Copyright (C) 2025 Wildfire Games.
  * This file is part of 0 A.D.
  *
  * 0 A.D. is free software: you can redistribute it and/or modify
@@ -25,6 +25,7 @@
 #include "ps/CStr.h"
 
 #include <algorithm>
+#include <cstring>
 #include <unordered_map>
 #include <vector>
 
-- 
2.49.0

