From 7c3d7c7b562d6efc6abeb99487efe3bb5cff922f Mon Sep 17 00:00:00 2001
From: Stan <stan@wildfiregames.com>
Date: Mon, 10 Feb 2025 21:24:51 +0100
Subject: [PATCH 08/10] Disable SpiderMonkey's debug build

---
 libraries/source/spidermonkey/build.sh | 18 ------------------
 1 file changed, 18 deletions(-)

diff --git a/libraries/source/spidermonkey/build.sh b/libraries/source/spidermonkey/build.sh
index 0dc66e0184..1c86ea4dbc 100755
--- a/libraries/source/spidermonkey/build.sh
+++ b/libraries/source/spidermonkey/build.sh
@@ -126,16 +126,6 @@ tar xfJ "${FOLDER}.tar.xz"
 		${CTARGET:+--target=${CTARGET}}"
 
 	# Build
-	# Debug (broken on FreeBSD)
-	if [ "${OS}" != "FreeBSD" ]; then
-		MOZCONFIG="$(pwd)/../mozconfig" \
-		MOZCONFIG_OPTIONS="${CONF_OPTS} \
-			--enable-debug \
-			--disable-optimize \
-			--enable-gczeal" \
-		BUILD_DIR="build-debug" \
-			./mach build "${JOBS}"
-	fi
 	# Release
 	MOZCONFIG="$(pwd)/../mozconfig" \
 	MOZCONFIG_OPTIONS="${CONF_OPTS} \
@@ -188,11 +178,6 @@ fi
 mkdir include-release
 cp -R -L "${FOLDER}"/build-release/dist/include/* include-release/
 
-if [ "${OS}" != "FreeBSD" ]; then
-	mkdir include-debug
-	cp -R -L "${FOLDER}"/build-debug/dist/include/* include-debug/
-fi
-
 # These align the ligns below, making it easier to check for mistakes.
 DEB="debug"
 REL="release"
@@ -220,9 +205,6 @@ elif [ "${OS}" = "Windows_NT" ]; then
 else
 	# Copy shared libs to lib/, they will also be copied to binaries/system, so the compiler and executable (resp.) can find them.
 	cp -L "${FOLDER}/build-${REL}/js/src/build/${LIB_PREFIX}${LIB_NAME}-${REL}${LIB_SUFFIX}" "lib/${LIB_PREFIX}${LIB_NAME}-${REL}${LIB_SUFFIX}"
-	if [ "${OS}" != "FreeBSD" ]; then
-		cp -L "${FOLDER}/build-${DEB}/js/src/build/${LIB_PREFIX}${LIB_NAME}-${DEB}${LIB_SUFFIX}" "lib/${LIB_PREFIX}${LIB_NAME}-${DEB}${LIB_SUFFIX}"
-	fi
 fi
 
 # cleanup
-- 
2.49.0

