Description: Add missing header in KeyName
Origin: upstream, https://gitea.wildfiregames.com/0ad/0ad/commit/bccd3db231ebf3d163f05c1996e48d2448acdbbe
Bug: https://gitea.wildfiregames.com/0ad/0ad/issues/7564
Author: Ralph Sennhauser <ralph.sennhauser@gmail.com>
Forwarded: not-needed
Last-Update: 2025-02-01

--- a/source/ps/KeyName.cpp
+++ b/source/ps/KeyName.cpp
@@ -1,4 +1,4 @@
-/* Copyright (C) 2021 Wildfire Games.
+/* Copyright (C) 2025 Wildfire Games.
  * This file is part of 0 A.D.
  *
  * 0 A.D. is free software: you can redistribute it and/or modify
@@ -25,6 +25,7 @@
 #include "ps/CStr.h"
 
 #include <algorithm>
+#include <cstring>
 #include <unordered_map>
 #include <vector>
 
